NXP/MCIMX8M_EVK_4GB/Library/MsPlatformDevicesLib/MsPlatformDevicesLib.c

```diff
@@ -0,0 +1,284 @@
+  { \
+    { \
+      HARDWARE_DEVICE_PATH, \
+      HW_PCI_DP, \
+      { \
+        (UINT8) ( sizeof (PCI_DEVICE_PATH)), \
+        (UINT8) ((sizeof (PCI_DEVICE_PATH)) >> 8), \
+      }, \
+    }, \
+    0x00, \
+    0x14 \
+  }
+
+#define gUsbClassKeyboard \
+  { \
+    { \
+      MESSAGING_DEVICE_PATH, \
+      MSG_USB_CLASS_DP, \
+      { \
+        (UINT8) ( sizeof (USB_CLASS_DEVICE_PATH)), \
+        (UINT8) ((sizeof (USB_CLASS_DEVICE_PATH)) >> 8), \
+      }, \
+    }, \
+    0xFFFF,                                  \
+    0xFFFF,                                  \
+    0x03,  /* HID Class */                   \
+    0x01,  /* Boot Interface Subclass  */    \
+    0x01   /* Keyboard Protocol */           \
+  }
+
+#define gUsbClassMouse \
+  { \
+    { \
+      MESSAGING_DEVICE_PATH, \
+      MSG_USB_CLASS_DP, \
+      { \
+        (UINT8) ( sizeof (USB_CLASS_DEVICE_PATH)), \
+        (UINT8) ((sizeof (USB_CLASS_DEVICE_PATH)) >> 8), \
+      }, \
+    }, \
+    0xFFFF,                                  \
+    0xFFFF,                                  \
+    0x03,  /* HID Class */                   \
+    0x01,  /* Boot Interface Subclass  */    \
+    0x02   /* Keyboard Protocol */           \
+  }
+
+typedef struct {
+  VENDOR_DEVICE_PATH        Guid;
+  UART_DEVICE_PATH          Uart;
+  EFI_DEVICE_PATH_PROTOCOL  End;
+} SIMPLE_TEXT_OUT_DEVICE_PATH;
+
+static SIMPLE_TEXT_OUT_DEVICE_PATH gUartDevicePath = {
+  {
+    { HARDWARE_DEVICE_PATH, HW_VENDOR_DP, {sizeof (VENDOR_DEVICE_PATH), 0}}
+  },
+  {
+    { MESSAGING_DEVICE_PATH, MSG_UART_DP,
+      {sizeof (UART_DEVICE_PATH), (UINT8) ((sizeof (UART_DEVICE_PATH)) >> 8)}},
+    0,        // Reserved
+    FixedPcdGet64 (PcdUartDefaultBaudRate),
+    FixedPcdGet8 (PcdUartDefaultDataBits),
+    FixedPcdGet8 (PcdUartDefaultParity),
+    FixedPcdGet8 (PcdUartDefaultStopBits)
+  },
+  gEndEntire
+};
+
+static GENERIC_USB_CLASS_DEVICE_PATH gUsbKeyboardDevices = {
+    gPciRootBridge,
+    gPciXhciDevice,
+    gUsbClassKeyboard,
+    gEndEntire };
+
+static GENERIC_USB_CLASS_DEVICE_PATH gUsbMouseDevices = {
+    gPciRootBridge,
+    gPciXhciDevice,
+    gUsbClassMouse,
+    gEndEntire };
+
+static EFI_DEVICE_PATH_PROTOCOL  *gPlatformConInDeviceList[] = {
+    (EFI_DEVICE_PATH_PROTOCOL *) &gUartDevicePath,
+    (EFI_DEVICE_PATH_PROTOCOL *) &gUsbKeyboardDevices,
+    (EFI_DEVICE_PATH_PROTOCOL *) &gUsbMouseDevices,
+    NULL
+};
+
+//
+// Predefined platform default console device path
+//
+static BDS_CONSOLE_CONNECT_ENTRY   gPlatformConsoles[] = {
+  //
+  // Place holder for serial console. Any non USB device used for
+  // CONIN must be in this table.  Any non display used for CONOUT
+  // must also be in this list.
+  //
+  {
+    (EFI_DEVICE_PATH_PROTOCOL *) &gUartDevicePath,
+    (CONSOLE_OUT | CONSOLE_IN)
+  },
+  {
+    NULL,
+    0
+  }
+};
+
+
+/**
+ * GetSdCardDevicePath
+ *
+ * Returns the device path of an internal SD Card slot that is
+ * excluded from being bootable.
```

Comment:
For iMX8 dev we boot from the SD card.  Is this comment accurate, or is it desired behaviour?